apps/store/src/components/Cancellation/CancellationForm.tsx

```diff
@@ -0,0 +1,55 @@
+import { InputStartDay } from '@/components/InputDay/InputStartDay'
+import {
+  type ProductOfferFragment,
+  useCancellationRequestedUpdateMutation,
+  usePriceIntentQuery,
+  useStartDateUpdateMutation,
+  ExternalInsuranceCancellationOption,
+} from '@/services/graphql/generated'
+import { convertToDate, formatAPIDate } from '@/utils/date'
+import { IEXCancellation } from './IEXCancellation'
+
+type Props = {
+  priceIntentId: string
+  offer: ProductOfferFragment
+}
+
+export const CancellationForm = (props: Props) => {
+  const { data } = usePriceIntentQuery({ variables: { priceIntentId: props.priceIntentId } })
+  if (!data) throw new Error('Cancellation | Missing data')
+
+  const productOfferIds = data.priceIntent.offers.map((item) => item.id)
+  const [mutate] = useCancellationRequestedUpdateMutation()
+  const handleAutoSwitchChange = (checked: boolean) => {
+    mutate({ variables: { productOfferIds, requested: checked } })
+  }
+
+  const startDate = convertToDate(props.offer.startDate)
+  const [updateStartDate] = useStartDateUpdateMutation()
+  const handleChangeStartDate = (date: Date) => {
+    updateStartDate({
+      variables: {
+        productOfferIds,
+        startDate: formatAPIDate(date),
+      },
+    })
+  }
+
```

Comment:
Question: I saw in `SwitchPage` that we deal with [only one offer](https://github.com/HedvigInsurance/racoon/pull/3795/files#diff-8db0f50cd06f1a4265d968d9c662f306f44592d48785e501466e3520a13b5269L36):

```
const offer = props.priceIntent.offers[0]
```

Which is then passed in to `CancellationForm` as the `offer` prop. Wouldn't be the case of changing `handleAutoSwitchChange` and `handleChangeStartDate` functions so they only mutation `props.offer.id`?

```suggestion

  const [mutate] = useCancellationRequestedUpdateMutation()
  const handleAutoSwitchChange = (checked: boolean) => {
    mutate({ variables: { productOfferIds: [props.offer.id], requested: checked } })
  }

  const startDate = convertToDate(props.offer.startDate)
  const [updateStartDate] = useStartDateUpdateMutation()
  const handleChangeStartDate = (date: Date) => {
    updateStartDate({
      variables: {
        productOfferIds: [props.offer.id],
        startDate: formatAPIDate(date),
      },
    })
  }

```

Reply:
good point, however, we want to target all offers (tiers) when updating any of these properties

perhaps we should not always take the first offer from the price intent though